### PR TITLE
fix golang-ci lint test

### LINKED ISSUE #7761: Fixing the CI issues
#### Current Behavior

This is the result of the `golangci-lint` process, we need to fix these issues below otherwise, the CI is not too much useful for reviewing the PR.

```
 Running [/home/runner/golangci-lint-1.52.2-linux-amd64/golangci-lint run --out-format=github-actions --timeout 5m --verbose] in [] ...
  Error: field `instancePath` is unused (unused)
  Error: field `schemaPath` is unused (unused)
  Error: field `propertyName` is unused (unused)
  Error: field `message` is unused (unused)
  Error: type `k8sMetadata` is unused (unused)
  Error: type `helperGV` is unused (unused)
  Error: type `kind` is unused (unused)
  Error: type `groupversion` is unused (unused)
  Error: func `getGroupsFromResource` is unused (unused)
  Error: func `(*DefaultLocalProvider).issueSession` is unused (unused)
  Error: func `downloadApplicationsFromURL` is unused (unused)
  Error: func `downloadYAMLSintoSingleFile` is unused (unused)
  Error: func `getSeededAppLocation` is unused (unused)
  Error: const `gRPC` is unused (unused)
  Error: const `istio` is unused (unused)
  Error: func `createIstioCanaryArgoResource` is unused (unused)
  Error: func `generateWorkloadKey` is unused (unused)
  Error: func `(*Selector).selectCoreWorkload` is unused (unused)
  Error: func `(*Selector).selectK8sWorkload` is unused (unused)
  Error: func `(*Selector).selectMeshWorkload` is unused (unused)
  Error: func `filterWorkloadByType` is unused (unused)
  Error: func `convertValueInterfaceSliceToWorkloadSlice` is unused (unused)
  Error: func `replaceParentInDependsOn` is unused (unused)
  Error: func `validateTrait` is unused (unused)
  Error: func `isLegalTrait` is unused (unused)
  Error: func `filterUniqueElementsArray` is unused (unused)
  Error: type `patternCallType` is unused (unused)
  Error: const `rawAdapter` is unused (unused)
  Error: const `noneLocal` is unused (unused)
  Error: const `oamAdapter` is unused (unused)
  Error: type `userInfo` is unused (unused)
  Error: type `users` is unused (unused)
  Error: type `usersParameterWrapper` is unused (unused)
  Error: type `mesheryPatternsResponseWrapper` is unused (unused)
  Error: type `mesheryPatternResponseWrapper` is unused (unused)
  Error: type `noContentWrapper` is unused (unused)
  Error: type `performanceProfilesResponseWrapper` is unused (unused)
  Error: type `performanceResultsResponseWrapper` is unused (unused)
  Error: type `performanceProfileResponseWrapper` is unused (unused)
  Error: type `performanceProfileParameterWrapper` is unused (unused)
  Error: type `performanceTestParameterWrapper` is unused (unused)
  Error: type `grafanaConfigParamsWrapper` is unused (unused)
  Error: type `grafanaConfigResponseWrapper` is unused (unused)
  Error: type `grafanaBoardsParamsWrapper` is unused (unused)
  Error: type `grafanaBoardsResponseWrapper` is unused (unused)
  Error: type `v1ServicesMapResponseWrapper` is unused (unused)
  Error: type `prometheusConfigParamsWrapper` is unused (unused)
  Error: type `prometheusConfigResponseWrapper` is unused (unused)
  Error: type `prometheusBoardImportRespWrapper` is unused (unused)
  Error: type `prometheusStaticBoardRespWrapper` is unused (unused)
  Error: S1039: unnecessary use of fmt.Sprintf (gosimple)
  Error: S1002: should omit comparison to bool constant, can be simplified to `gpi.Register` (gosimple)
  Error: S1011: should replace loop with `selectors = append(selectors, model.AllMeshType...)` (gosimple)
  Error: S1011: should replace loop with `selectors = append(selectors, model.AllMeshType...)` (gosimple)
  Error: S1006: should use for {} instead of for true {} (gosimple)
  Error: S1023: redundant `return` statement (gosimple)
  Error: S1023: redundant `return` statement (gosimple)
  Error: S1023: redundant `return` statement (gosimple)
  Error: S1040: type assertion to the same type: err already has type error (gosimple)
  Error: S1040: type assertion to the same type: err already has type error (gosimple)
  Error: S1019: should use make(chan []*model.AddonList) instead (gosimple)
  Error: S1025: the argument is already a string, there's no need to use fmt.Sprintf (gosimple)
  Error: S1009: should omit nil check; len() for []github.com/layer5io/meshkit/models/meshmodel.Entity is defined as zero (gosimple)
  Error: S1021: should merge variable declaration with assignment on next line (gosimple)
  Error: S1008: should use 'return len(ss.sws) == 0' instead of 'if len(ss.sws) == 0 { return true }; return false' (gosimple)
  Error: ineffectual assignment to step (ineffassign)
  Error: ineffectual assignment to version (ineffassign)
  Error: ineffectual assignment to expectedResponseForAll (ineffassign)
  Error: ineffectual assignment to expectedResponseForAll (ineffassign)
```

#### Environment

- **Host OS:** Mac Linux Windows
- **Platform:** Docker or Kubernetes
- **Meshery Server Version:** stable-v
- **Meshery Client Version:** stable-v

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by ayushthe1**
**Notes for Reviewers**

This PR fixes #7761 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by ayushthe1** [comment]
@theBeginner86 @Aisuko I haven't removed the unused functions from `docs.go` file as you discussed. Thats why the test are failing. I'll exclude it from the lint tests. Meanwhile please review this PR as I have removed many more unused  functions which might be needed.

File: server/models/meshmodel/core/register.go [hunk 1]
```diff
@@ -323,55 +317,55 @@ type kind string
 type groupversion string
```
Comment (Aisuko) [inline server/models/meshmodel/core/register.go:317]: Are these structs still useful?

File: server/models/meshmodel/core/register.go [hunk 2]
```diff
@@ -323,55 +317,55 @@ type kind string
...
-	var gl v1.APIGroupList
-	gs, err := cli.KubeClient.RESTClient().Get().RequestURI("/apis").Do(context.Background()).Raw()
-	if err != nil {
-		return nil, err
-	}
-	err = json.Unmarshal(gs, &gl)
-	if err != nil {
-		return nil, err
-	}
+// func getGroupsFromResource(cli *kubernetes.Client) (hgv map[kind][]groupversion, err error) {
```
Comment (Aisuko) [inline server/models/meshmodel/core/register.go:320]: Just remove it?

**Review by Aisuko — APPROVED** [review]
LGTM

**Pull request merged**